editor/editor_node.cpp

```diff
@@ -3697,21 +3697,28 @@ StringName EditorNode::get_object_custom_type_name(const Object *p_object) const
 	return StringName();
 }
 
-Ref<ImageTexture> EditorNode::_load_custom_class_icon(const String &p_path) const {
-	if (p_path.length()) {
-		Ref<Image> img = memnew(Image);
-		Error err = ImageLoader::load_image(p_path, img);
-		if (err == OK) {
-			Ref<ImageTexture> icon = memnew(ImageTexture);
-			img->resize(16 * EDSCALE, 16 * EDSCALE, Image::INTERPOLATE_LANCZOS);
-			icon->create_from_image(img);
-			return icon;
+Ref<ImageTexture> EditorNode::_load_custom_class_icon(const String &p_path) {
+	// FIXME: For some reason, p_path is sometimes the literal string "Null" rather than an empty string
+	if (p_path.length() && p_path != "Null") {
+		Error err;
+		if(ResourceLoader::get_resource_type(p_path) == "Image") {
```

Comment:
Why change to `ResourceLoader` instead of `ImageLoader::load_image`? It's better not to have to use a string literal and simply load the type we want, or return an error if it fails.

Or does `load_image` raise error messages if its input is not an image?

Reply:
Ah I saw your comments on the PR about using `ResourceLoader` for caching, I guess that makes sense.